gce: support assign interface of local ssd

amoskong opened the pull request:
GCE has two kind of interface for local ssd, NVME and SCSI.
We always use NVME in current code, this patch added support
to assign interface by `gce_local_ssd_if_db/loader/monitor`.

We will still use NVME by default.

amoskong commented:
I will rebase it.

On Thu, Jan 25, 2018 at 12:50 AM, Larisa Ustalov <notifications@github.com>
wrote:

> *@larisau* commented on this pull request.
>
> this PR conflicts with #469
> <https://github.com/scylladb/scylla-cluster-tests/pull/469> , need to
> rebase
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/scylladb/scylla-cluster-tests/pull/478#pullrequestreview-91247763>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAS5zC1iFJBzaoM5ilLXnrN47L37NELSks5tN186gaJpZM4RjGvX>
> .
>

larisau commented on sdcm/cluster_gce.py:
```diff
@@ -181,6 +182,8 @@ def _get_root_disk_struct(self, name, disk_type='pd-standard', dc_idx=0):
     def _get_local_ssd_disk_struct(self, name, index, interface='NVME', dc_idx=0):
         device_name = '%s-data-local-ssd-%s' % (name, index)
+        if local_ssd_if is None:
+            local_ssd_if = 'NVME'
```
the variable local_ssd_if is not used

larisau commented on sdcm/cluster_gce.py:
```diff
@@ -127,7 +127,7 @@ class GCECluster(cluster.BaseCluster):
                  cluster_uuid=None, gce_instance_type='n1-standard-1', gce_region_names=['us-east1-b'],
                  gce_n_local_ssd=1, gce_image_username='root', cluster_prefix='cluster',
-                 node_prefix='node', n_nodes=[10], add_disks=None, params=None):
+                 node_prefix='node', n_nodes=[10], add_disks=None, params=None, gce_local_ssd_if=None):
```
I think it's better to set default value as 'NVME':
`gce_local_ssd_if='NVME'`

amoskong commented:
I have updated the PR to set default gce_local_ssd_if to None, and
removed `local_ssd_if`
usage inside _get_local_ssd_disk_struct(). Thanks Larisa.

On Sun, Jan 28, 2018 at 4:44 PM, Larisa Ustalov <notifications@github.com>
wrote:

> *@larisau* commented on this pull request.
> ------------------------------
>
> In sdcm/cluster_gce.py
> <https://github.com/scylladb/scylla-cluster-tests/pull/478#discussion_r164293187>
> :
>
> > @@ -181,6 +182,8 @@ def _get_root_disk_struct(self, name, disk_type='pd-standard', dc_idx=0):
>
>      def _get_local_ssd_disk_struct(self, name, index, interface='NVME', dc_idx=0):
>          device_name = '%s-data-local-ssd-%s' % (name, index)
> +        if local_ssd_if is None:
> +            local_ssd_if = 'NVME'
>
> the variable local_ssd_if is not used
> ------------------------------
>
> In sdcm/cluster_gce.py
> <https://github.com/scylladb/scylla-cluster-tests/pull/478#discussion_r164293211>
> :
>
> > @@ -127,7 +127,7 @@ class GCECluster(cluster.BaseCluster):
>      def __init__(self, gce_image, gce_image_type, gce_image_size, gce_network, services, credentials,
>                   cluster_uuid=None, gce_instance_type='n1-standard-1', gce_region_names=['us-east1-b'],
>                   gce_n_local_ssd=1, gce_image_username='root', cluster_prefix='cluster',
> -                 node_prefix='node', n_nodes=[10], add_disks=None, params=None):
> +                 node_prefix='node', n_nodes=[10], add_disks=None, params=None, gce_local_ssd_if=None):
>
> I think it's better to set default value as 'NVME':
> gce_local_ssd_if='NVME'
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/scylladb/scylla-cluster-tests/pull/478#pullrequestreview-92046758>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAS5zP8L1CELwhkDaBYhkOVkhuWm9xnPks5tPDNagaJpZM4RjGvX>
> .
>

roydahan commented:
@amoskong, I don't understand why we need this PR...
What functionality does it add or what does it fix?

IIUC, before this PR, if the yaml had "0" for n_local_ssd" and has gce_pd_ssd_disk_size_db or gce_pd_standard_disk_size_db the behaviour will be the same.

Although we have "node.remoter.run('ls /dev/nvme0n*')" if we don't have any local_ssd (n_local_ssd: 0), it won't fail.

Am I wrong?

amoskong commented:
Local ssd have two kind of interface: NVMe or SCSI.

When we use SCSI, the device file name is /dev/sd*, "node.remoter.run('ls
/dev/nvme0n*')" should not fail.

On Mon, Jan 29, 2018 at 9:42 PM, Roy Dahan <notifications@github.com> wrote:

> @amoskong <https://github.com/amoskong>, I don't understand why we need
> this PR...
> What functionality does it add or what does it fix?
>
> IIUC, before this PR, if the yaml had "0" for n_local_ssd" and has
> gce_pd_ssd_disk_size_db or gce_pd_standard_disk_size_db the behaviour will
> be the same.
>
> Although we have "node.remoter.run('ls /dev/nvme0n*')" if we don't have
> any local_ssd (n_local_ssd: 0), it won't fail.
>
> Am I wrong?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/scylladb/scylla-cluster-tests/pull/478#issuecomment-361248607>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAS5zM2QoTM5jrjTTo8UVtYAF55_VUKvks5tPcq5gaJpZM4RjGvX>
> .
>

larisau commented:
@amoskong The test working with pd_standard doesn't fail on any of 'ls' commands, you can see the device /dev/sdb, and there is no interface choice for it, it's always SCSI.
It fails on scylla_setup.

roydahan commented:
So, this PR is to support local_ssd with SCSI interface?

roydahan commented:
@amoskong, there are some conflicts.
However, I'm not sure we really need this PR.
if we need SCSI disk we can use pd-ssd or pd-standard, I don't see the use case of using local_ssd with SCSI interface.

If you think there is a use case for it, fix the conflicts and merge.

amoskong commented:
I don't have usage for this new interface, let's close this PR.
